### ISSUE TITLE
Feedback on summing P(z) thoughts

### ISSUE THREAD

**Issue by drphilmarshall** [danielgruen/sumzz#9]
Hi @aimalz ! Here's the project I was telling you about last week at the LSST DESC Hack Week. Interested to hear your thoughts on our [notes](https://github.com/danielgruen/sumzz/blob/pdf/sumzz.pdf) - marginalizing over datasets was a new concept to me, and seems like an interesting thing to think about.

**Comment by drphilmarshall** [comment]
@danielgruen Alex is here visiting us this summer! His arrival prompted a short conversation with @davidwhogg, who prompted me to revisit this note that we wrote. Looking back at what we wrote about the stacked estimator, I think our conclusions section doesn't match the main text in terms of the strength of our conclusions about the stacked estimator. For example, in our conclusions, we wrote: 

> "If you have a machine learning photometric redshift code that
returns individual posterior PDFs $P(z_i | d_i, I)$ by interpolating
in data space over a large and representative training set, then an
approximation to the underlying redshift distribution $P(z | T)$ is
given by the stacked estimator $\frac{1}{N_g} \sum_{i=1}^{N_g}
P(z_i | d_i, I)$."

I think we should point out that one resulting research question is whether this training set can ever be made large enough or representative enough to give accurate results in the inferred hyperparameters (like the cosmological ones). 

Then, in the section on "Stacking despite model uncertainty" we wrote some stronger words:

> "What if we assume a model $H$ that is not the true model $T$? 
We can write down $P(a|h,H)$ and $P(h|H)$, and the individual galaxy posterior PDF becomes
\begin{align}
P(z_i | d_i, H) &= \int\int P(z_i|a_i) P(a_i|d_i,h,H) P(h|H) \; da_i\; dh \\
&= \int\int P(z_i|a_i) P(a_i|h,H) P(h|H) P(d_i|a_i,h,H) / P(d_i|H) \; da_i \; dh \; .
\end{align}
Because $P(a,h)=P(a|h)P(h)$ and $P(d_i|a,h)$ depend on $h$, this PDF
will actually be broader than the distribution of $z$ that we would get
if we binned up the true redshifts of all galaxies that look like $d_i$."

But this point does not make it into the conclusions. I'll prepare a quick PR to address this.

**Comment by davidwhogg** [comment]
I have a *stronger* objection: If the training set is good enough to be representative, IT is your measure of N(z). Or, another way to put it: Your argument that the stacked posterior gives an unbiased N(z) is true even if the data provide NO INFORMATION: That is, if the likelihoods are null, the posterior equals the prior, and bang! you get the right N(z). So this is just a statement that your prior is correct. In that case, you don't need data!

Further argument: If your likelihoods are not null, so you *are* getting information from them, then they actually *noisify* rather than *improve* your N(z) estimate. They make it worse. That is, it is worse to stack posteriors than to just use your assumed-correct training-set N(z).

**Comment by aimalz** [comment]
Yes, this was what I was trying to say yesterday; 1) stacking always biases you by the interim prior, which is always problematic except if your interim prior is the truth and 2) stacking posteriors will always broaden the recovered distribution even if your interim prior is the truth.

**Comment by davidwhogg** [comment]
I think that maybe (2) is wrong: It might be true that if the interim prior is exactly right, then stacking posteriors in the end will asymptote to a correct N(z).

But not for any inferential reason! That is, not because the *data* are contributing *information*.

**Comment by danielgruen** [comment]
I agree with your comments, @aimalz and @davidwhogg -- I guess I was just coming from an old-school photo-z world in which the interim prior is sort of authoritatively assumed to be exactly right.

The fact that the simple stacking is no longer correct if the data constrain the interim prior is a valuable insight, at least to me. We need to make sure that's reflected in the note.

I merged Phil's pull request, which goes in that direction. But maybe we should put this even more explicitly. Do you want to have a look?
